src/bp-forums/forums/functions.php

```diff
@@ -2563,3 +2575,28 @@ function bbp_get_forums_per_page( $default = 15 ) {
 	// Filter and return
 	return (int) apply_filters( 'bbp_get_forums_per_page', $retval, $default );
 }
+
+/**
+ * Redirect non-loggedin user to login page while viewing private forum
+ *
+ * @param object $posts array of post objects.
+ * @param object $wp_query WP_Query inatsnace.
+ */
+function bbp_redirect_private_forum( $posts, &$wp_query ) {
+	// remove filter now, so that on subsequent post querying we don't get involved!
+	remove_filter( 'the_posts', 'bbp_redirect_private_forum', 5, 2 );
+
+	$forum_slug = get_query_var( 'forum' );
```

Comment:
@ngolwalkar What happens if someone has another slug instead of the `forum`?

Reply:
It will still work as the query var still be the forum.